### PR TITLE
Support v3io over-time aggregations

### PR DESCRIPTION
**Pull request by talIguaz**
* Bump tsdb to 0.9
* support v3io aggregations for `<aggr>_over_time`

### CODE REVIEW

File: promql/engine.go [hunk 1]
```diff
@@ -529,6 +531,7 @@ func (ng *Engine) populateSeries(ctx context.Context, q storage.Queryable, s *Ev
 				maxOffset = n.Offset + LookbackDelta + subqOffset
 			}
 		case *MatrixSelector:
+			aggregationWindow = n.Range.Nanoseconds() / 1000000
```
Comment (gtopper) [inline promql/engine.go]: `1000000` -> `time.Millisecond`?

File: promql/engine.go [hunk 2]
```diff
@@ -1910,21 +1916,44 @@ func isV3ioEligibleAggregation(op ItemType) bool {
...
 }
 
-func isV3ioEligibleQueryExpr(e *AggregateExpr) bool {
-	if !isV3ioEligibleAggregation(e.Op) {
-		return false
-	}
-	if e.Without {
-		return false
+func isV3ioEligibleFunction(function string) bool {
+	supportedV3ioAggregations := []string{"max_over_time",
```
Comment (gtopper) [inline promql/engine.go]: Maybe make this a set?

File: storage/tsdb/promtsdb.go [hunk 3]
```diff
@@ -105,23 +105,28 @@ func (promQuery *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*l
...
 		// TODO: support count/stdxx, require changes in Prometheus: promql/functions.go, not calc aggregate twice
 		if strings.HasSuffix(params.Func, "_over_time") {
-			f := params.Func[0:3]
-			if params.Step == 0 && (f == "min" || f == "max" || f == "sum" || f == "avg") {
-				functions = f
+			if promQuery.UseAggregates && promQuery.UseAggregatesConfig {
+				functions = params.Func[0:3]
 			} else {
-				noAggr = true
+				f := params.Func[0:3]
```
Comment (gtopper) [inline storage/tsdb/promtsdb.go]: @talIguaz says this is wrong.

File: promql/engine.go (hunk 1 above)
Comment (talIguaz) [reply promql/engine.go]: time.Millisecond is Duration which means ill have to cast it again..

File: storage/tsdb/promtsdb.go [hunk 4]
```diff
@@ -100,28 +100,34 @@ func (promQuery *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*l
...
-			f := params.Func[0:3]
-			if params.Step == 0 && (f == "min" || f == "max" || f == "sum" || f == "avg") {
-				functions = f
+		if strings.HasSuffix(params.Func, overTimeSuffix) {
+			if promQuery.UseAggregates && promQuery.UseAggregatesConfig {
+				functions = strings.TrimSuffix(params.Func, overTimeSuffix)
 			} else {
-				noAggr = true
+				f := params.Func[0:3]
+				if params.Step == 0 && (f == "min" || f == "max" || f == "sum" || f == "avg") {
```
Comment (gtopper) [inline storage/tsdb/promtsdb.go]: Not really safe... Could be called `summary`, and that starts with `sum`... But not a regression in this PR.

File: storage/tsdb/promtsdb.go [hunk 5]
```diff
@@ -100,28 +100,34 @@ func (promQuery *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*l
...
 	if params.Func != "" {
 		// only pass xx_over_time functions (just the xx part)
 		// TODO: support count/stdxx, require changes in Prometheus: promql/functions.go, not calc aggregate twice
-		if strings.HasSuffix(params.Func, "_over_time") {
-			f := params.Func[0:3]
-			if params.Step == 0 && (f == "min" || f == "max" || f == "sum" || f == "avg") {
-				functions = f
+		if strings.HasSuffix(params.Func, overTimeSuffix) {
+			if promQuery.UseAggregates && promQuery.UseAggregatesConfig {
+				functions = strings.TrimSuffix(params.Func, overTimeSuffix)
```
Comment (gtopper) [inline storage/tsdb/promtsdb.go]: Actually a single function, but not a regression in this PR.

**Pull request merged**